### PR TITLE
Adding postgres docker to jenkins test

### PR DESCRIPTION
**Pull request by Samyoul**
## What's Changed?

I've eventually added postgres support into the CI testing. This approach doesn't need to use a Docker hostname like `db` and defaults to using the '127.0.0.1' host address for the Postgres DB, which means that we don't need to update the code to run the go tests in CI or locally on the user's machine.

If running the full suite of tests on the local machine or working on tests that require Postgres the following command should be executed (Docker is required to be installed on your local machine):

```sh
docker run --name test-db -e POSTGRES_HOST_AUTH_METHOD=trust -d -p 5432:5432 postgres:9.6-alpine
```

## Why Change?

We have tests that require a postgres db but we don't have a way to successfully test these in CI. Now we do 😎 

Now the following tests run and pass:

```text
[2021-06-08T15:31:37.449Z] === RUN   TestPostgresDB_BuildIteratorWithBloomFilter
[2021-06-08T15:31:37.449Z] --- PASS: TestPostgresDB_BuildIteratorWithBloomFilter (0.04s)
[2021-06-08T15:31:37.449Z] === RUN   TestPostgresDB_BuildIteratorWithTopic
[2021-06-08T15:31:37.449Z] --- PASS: TestPostgresDB_BuildIteratorWithTopic (0.04s)
```

### CODE REVIEW

**Comment by jakubgs** [comment]
This works: https://github.com/status-im/status-go/commit/4b5ae65c
```Jenkinsfile
    stage('Unit Tests') { steps { script { dir(env.STATUS_PATH) {
      /* Some tests require PostgreSQL database. */
      docker.image('postgres:9.6-alpine').withRun([
        '--name mailserver-db',
        '-e POSTGRES_HOST_AUTH_METHOD=trust',
        '-d -p 5432:5432',
      ].join(' ')) { c ->
        /* Wait until postgres service is up. */
        sh 'timeout 10 sh -c "until nc -z localhost 5432; do sleep 0.2; done"'
        sh 'make test-unit'
      }
    } } } }
```
Tested here: https://ci.status.im/job/status-go/job/prs/job/tests/job/test-postgres-db/

Actually, I'm not even sure the `netcat` wait is even necessary. PG starts fast, and Go tests take a bit to start too.

**Comment by jakubgs** [comment]
I think we can also drop `--name mailserver-db` as Jenkins cleans up the container anyway, and `-d` is added by jenkins too.

**Comment by jakubgs** [comment]
So it could be as simple as:
```Jenkinsfile
    stage('Unit Tests') { steps { script { dir(env.STATUS_PATH) {
      docker.image('postgres:9.6-alpine').withRun(
        '-e POSTGRES_HOST_AUTH_METHOD=trust -p 5432:5432'
      ) { c ->
        sh 'make test-unit'
      }
    } } } }
```

**Comment by jakubgs** [comment]
I don't think this is gonna fly with `agent none` my dude. The actual error is:
```
nohup: failed to run command 'sh': No such file or directory
```
You can probably drop all of the changes aside from the ones in `Unit Tests`.

**Review by cammellos — APPROVED** [review]

File: _assets/ci/Jenkinsfile.tests [hunk 1]
```diff
@@ -54,9 +54,13 @@ pipeline {
       sh 'make canary-test'
     } } }
 
-    stage('Unit Tests') { steps { dir(env.STATUS_PATH) {
-      sh 'make test-unit'
-    } } }
+    stage('Unit Tests') { steps { script { dir(env.STATUS_PATH) {
+      docker.image('postgres:9.6-alpine').withRun(
+              '-e POSTGRES_HOST_AUTH_METHOD=trust -p 5432:5432'
```
Comment (jakubgs) [inline _assets/ci/Jenkinsfile.tests]: Just to be extra annoying: wrong indentation.

**Pull request merged**